src/SFA.DAS.EmployerAccounts/Services/CommitmentsV2ServiceWithTimeout.cs

```diff
@@ -0,0 +1,113 @@
+        {
+            try
+            {
+                var draftApprenticeshipsResponse = await _pollyPolicy.ExecuteAsync(() => _commitmentsApiClient.GetDraftApprenticeships(cohort.Id));
+                return _mapper.Map<IEnumerable<DraftApprenticeshipDto>, IEnumerable<Apprenticeship>>(draftApprenticeshipsResponse.DraftApprenticeships,
+                    opt =>
+                    {
+                        opt.AfterMap((src, dest) =>
+                        {
+                            dest.ToList().ForEach(c =>
+                            {
+                                c.SetHashId(_encodingService);
+                                c.SetCohort(cohort);
+                                c.SetTrainingProvider(cohort.TrainingProvider.First());
+                            });
+                        });
+                    });
+            }
+            catch (TimeoutRejectedException ex)
+            {
+                throw new ServiceTimeoutException("Call to Commitments V2 Service timed out", ex);
+            }
+        }
+
+        public async Task<IEnumerable<Cohort>> GetCohorts(long? accountId)
+        {
+            try
+            {
+                var cohortSummary = await _pollyPolicy.ExecuteAsync(() => _commitmentsApiClient.GetCohorts(new CommitmentsV2.Api.Types.Requests.GetCohortsRequest { AccountId = accountId }));
+                var trainingProvider = _mapper.Map<IEnumerable<CohortSummary>, IEnumerable<TrainingProvider>>(cohortSummary.Cohorts);
+
+                return _mapper.Map<IEnumerable<CohortSummary>, IEnumerable<Cohort>>(cohortSummary.Cohorts,
+                    opt =>
+                    {
+                        opt.AfterMap((src, dest) =>
+                        {
+                            dest.ToList().ForEach(c =>
+                            {
+                                c.SetHashId(_encodingService);
+                                c.SetTrainingProvider(trainingProvider);
+                            });
+                        });
+                    });
+            }
+            catch (TimeoutRejectedException ex)
+            {
+                throw new ServiceTimeoutException("Call to Commitments V2 Service timed out", ex);
+            }
+        }
+
+        public async Task<IEnumerable<Apprenticeship>> GetApprenticeships(long accountId)
+        {
+            try
+            {
+                var apprenticeship = await _pollyPolicy.ExecuteAsync(() => _commitmentsApiClient.GetApprenticeships(new CommitmentsV2.Api.Types.Requests.GetApprenticeshipsRequest { AccountId = accountId }));
+                var trainingProvider = _mapper.Map<IEnumerable<GetApprenticeshipsResponse.ApprenticeshipDetailsResponse>, IEnumerable<TrainingProvider>>(apprenticeship.Apprenticeships);
+
+                return _mapper.Map<IEnumerable<GetApprenticeshipsResponse.ApprenticeshipDetailsResponse>, IEnumerable<Apprenticeship>>(apprenticeship.Apprenticeships,
```

Comment:
This mapper shouldn't be in here.  It's nothing to do with the Timeout handling.